Mathlib/Data/Set/ToFunction.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Data.Set.Defs
+import Mathlib.Logic.ExistsUnique
+
+/-!
+# Set to function
+
+This file provides API for sets that define functions.
+
+## Main definitions
+- `Set.asPartialFun` converts `Set (α × α)` to `α → Option α`.
```

Comment:
Why not convert `Set (α × β)` into `α → Option β` / `α → β` instead?

Reply:
Good idea! Thanks!